metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -202,6 +203,26 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int) ([]string,
 	return result, true
 }
 
+func (m *metricHints) checkHostPort(h string, p int) bool {
+	port := strconv.Itoa(p)
+
+	index := strings.LastIndex(h, ":"+port)
+	// Check if host contains :port. If not then return false
+	if index == -1 {
+		return false
+	}
+
+	// Check if the host ends with :port. Return true if yes
+	end := index + len(port) + 1
+	if end == len(h) {
+		return true
+	}
+
+	// Check if the character immediately after :port. If its not a number then return true.
```

Comment:
I would suggest adding the `"tcp(${data.host}:3306)/"` example in this comment for better understanding and readability in the future.

Reply:
done